pkg/controller/vitessshard/reconcile_tablets.go

```diff
@@ -360,3 +362,45 @@ func tabletAvailableStatus(resultBuilder *results.Builder, pod *corev1.Pod, read
 	resultBuilder.RequeueAfter(tabletAvailableTime - readyDuration)
 	return corev1.ConditionFalse
 }
+
+func (r *ReconcileVitessShard) checkPVCDiskUpdates(ctx context.Context, tabletSpec *vttablet.Spec, pod *corev1.Pod) {
+	// If no PVC is configured for this tablet pod, bail out.
+	if tabletSpec.DataVolumePVCSpec == nil {
+		return
+	}
+
+	pvc := &corev1.PersistentVolumeClaim{}
+	namespacedClaim := apitypes.NamespacedName{
+		Namespace: pod.Namespace,
+		Name:      tabletSpec.DataVolumePVCName,
+	}
+
+	// If a matching PVC doesn't exist for this tablet pod, bail out.
+	err := r.client.Get(ctx, namespacedClaim, pvc)
+	if err != nil {
+		return
+	}
+
+	// If the matching PVC does not have the FileSystemResizePending condition, bail out.
+	if !checkPVCFileSystemResizeCondition(pvc) {
+		return
+	}
+
+	// If all checks pass, set the resize annotation.
+	requestedDiskQuantity := tabletSpec.DataVolumePVCSpec.Resources.Requests[corev1.ResourceStorage]
```

Comment:
We should also bail if the actual PVC's Spec size has not yet been updated to be equal to the desired size in DataVolumePVCSpec. As mentioned elsewhere, we should bail if DataVolumePVCSpec has a missing ResourceStorage entry as well.

Reply:
Sounds good, done. Going by the logic you gave me before, I moved the check for the `FileSystemResizeCondition` after checking that the PVC Spec size equals the desired size. Let me know if this logic doesn't hold here.